FreeRTOS-Plus/Demo/MQTT_Mutual_Auth_Qemu_mps2/main_networking.c

```diff
@@ -0,0 +1,338 @@
+                                    configMAC_ADDR3,
+                                    configMAC_ADDR4,
+                                    configMAC_ADDR5
+                                  };
+
+/* Set the following constant to pdTRUE to log using the method indicated by the
+ * name of the constant, or pdFALSE to not log using the method indicated by the
+ * name of the constant.  Options include to standard out (xLogToStdout), to a disk
+ * file (xLogToFile), and to a UDP port (xLogToUDP).  If xLogToUDP is set to pdTRUE
+ * then UDP messages are sent to the IP address configured as the echo server
+ * address (see the configECHO_SERVER_ADDR0 definitions in FreeRTOSConfig.h) and
+ * the port number set by configPRINT_PORT in FreeRTOSConfig.h. */
+//const BaseType_t xLogToStdout = pdTRUE, xLogToFile = pdFALSE, xLogToUDP = pdFALSE;
+
+
+/* Use by the pseudo random number generator. */
+static UBaseType_t ulNextRand;
+
+/*-----------------------------------------------------------*/
+
+void vMainMQTTClientTasks( void )
+{
+    const uint32_t ulLongTime_ms = pdMS_TO_TICKS( 1000UL );
+
+    /*
+     * Instructions for using this project are provided on:
+     * http://www.freertos.org/FreeRTOS-Plus/FreeRTOS_Plus_TCP/examples_FreeRTOS_simulator.html
+     */
+
+    /* Miscellaneous initialisation including preparing the logging and seeding
+     * the random number generator. */
+    prvMiscInitialisation();
+
+    /* Initialise the network interface.
+     *
+     ***NOTE*** Tasks that use the network are created in the network event hook
+     * when the network is connected and ready for use (see the definition of
+     * vApplicationIPNetworkEventHook() below).  The address values passed in here
+     * are used if ipconfigUSE_DHCP is set to 0, or if ipconfigUSE_DHCP is set to 1
+     * but a DHCP server cannot be contacted. */
+    FreeRTOS_debug_printf( ( "FreeRTOS_IPInit\n" ) );
+    FreeRTOS_IPInit( ucIPAddress,
+                     ucNetMask,
+                     ucGatewayAddress,
+                     ucDNSServerAddress,
+                     ucMACAddress );
+
+    /* Start the RTOS scheduler. */
+    FreeRTOS_debug_printf( ( "vTaskStartScheduler\n" ) );
+    vTaskStartScheduler();
+    FreeRTOS_debug_printf( ( "Should not reach this point after scheduler\n" ) );
+
+    /* If all is well, the scheduler will now be running, and the following
+     * line will never be reached.  If the following line does execute, then
+     * there was insufficient FreeRTOS heap memory available for the idle and/or
+     * timer tasks to be created.  See the memory management section on the
+     * FreeRTOS web site for more details (this is standard text that is not not
+     * really applicable to the Linux simulator port). */
+    for( ; ; )
+    {
+        usleep( ulLongTime_ms * 1000 );
+    }
+}
+/*-----------------------------------------------------------*/
+
+    BaseType_t xTasksAlreadyCreated = pdFALSE;
```

Comment:
why is this tabbed over in the middle here? move up to be with other globals and add a comment for it

Reply:
Fixed